src/main/js/ephox/imagetools/transformations/ImageTools.js

```diff
@@ -80,13 +82,13 @@ define("ephox/imagetools/transformations/ImageTools", [
 
   function resize(blob, w, h) {
     return Conversions.blobToImage(blob).then(function(image) {
-      var canvas = Canvas.create(w, h),
-        context = Canvas.get2dContext(canvas);
-
-      context.drawImage(image, 0, 0, w, h);
-      revokeImageUrl(image);
-
-      return Conversions.canvasToBlob(canvas, blob.type);
+      return ImageResizerWebgl.scale(image, w, h)
```

Comment:
The FP approach spocke is talking about would be to have a function that takes a resize implementation, an image and width/height then handles all of the scaffolding such as converting to and from canvas. Then you pass that function either `ImageResizerWebgl` or `ImageResizerCanvas` as appropriate.

That approach would require more proactive switching to canvas here rather than the try/catch style, but it generally leads to cleaner code.